src/workerd/api/r2-multipart.c++

```diff
@@ -137,7 +162,11 @@ jsg::Promise<jsg::Ref<R2Bucket::HeadResult>> R2MultipartUpload::complete(jsg::Lo
     KJ_IF_SOME(b, this->bucket->bucketName()) {
       traceContext.userSpan.setTag("cloudflare.r2.bucket"_kjc, kj::str(b));
     }
-    traceContext.userSpan.setTag("cloudflare.r2.upload_id"_kjc, kj::str(uploadId));
+    traceContext.userSpan.setTag("cloudflare.r2.request.upload_id"_kjc, kj::str(uploadId));
+    traceContext.userSpan.setTag("cloudflare.r2.request.key"_kjc, kj::str(key));
+    kj::String partIds =
+        kj::strArray(KJ_MAP(part, uploadedParts) { return kj::str(part.partNumber); }, ", ");
+    traceContext.userSpan.setTag("cloudflare.r2.request.uploaded_parts"_kjc, kj::str(partIds));
```

Comment:
nit: it doesn't really matter at this stage but just something to keep in mind for development:  Since partIds is already a kj::String and is not needed elsewhere, you can use `kj::mv(partIds)` here to avoid a memory allocation.